### ISSUE TITLE
UserWarning: y_pred contains classes not in y_true   warnings.warn("y_pred contains classes not in y_true")

### ISSUE THREAD

**Issue by upper127** [935963004/LaBraM#22]
Test:  [7/8]  eta: 0:00:06  loss: 1.3861 (1.3866)  accuracy: 0.2396 (0.2500)  balanced_accuracy: 0.2500 (0.2471)  cohen_kappa: 0.0000 (0.0000)  f1_weighted: 0.0926 (0.1013)  time: 6.2649  data: 4.9884  max mem: 6735
Test: Total time: 0:00:51 (6.4069 s / it)
* loss 1.387
Accuracy of the network on the 680 test EEG: 0.25%
Max accuracy val: 0.25%, max accuracy test: 0.25%
I can't seem to see the effect of the training.

**Comment by upper127** [comment]
This problem occurs with the TUEV dataset, and replacing my own dataset also

**Comment by wenjiu100** [comment]
I have the same problem with TUEV, even if I change the batchsize to 512. the result is still very bad

**Comment by samin9796** [comment]
@upper127 @wenjiu100 Hi! I am using my own dataset and the result is very bad since the loss does not converge. Did you able to get good performance on TUEV later? Also, did you have to fill up the form to get the TUEV dataset?

Thank you. I am looking forward to your reply.

**Comment by upper127** [comment]
Sorry for resuming your message so late.So far I haven't had a good result using my own dataset for validation.There are also some minor issues with using the TUEV dataset, and I'm not sure what's wrong. I can tell you for sure that if you want access to the TUEV or TUAB data, you have to fill out the form to request access.

**Comment by samin9796** [comment]
@upper127 Thank you for the information. I have access to the Temple University hospital EEG dataset, but don't know how to download them. If possible, do you have any shareable script to download them from the server?

**Comment by wenjiu100** [comment]
@samin9796 ,Sorry to get back to you now, using the TUEV dataset later turned out to be pretty good, pretty much the same as in his article, had to submit the form to request an account to use TUEV

**Comment by wenjiu100** [comment]
Log in to their website, which has a tutorial on how to download the data, and so far it looks like the only way to do it is to use a linux system

**Comment by Zhreyu** [comment]
@wenjiu100 hello, could you tell me how you were able to get better results later?

**Comment by wenjiu100** [comment]
@Zhreyu The data processing of tuev is different from tuab, you have to carefully analyze the code given by the authors, and you have to use the pre-training weights, so that it will be fine

**Comment by LwkBuaa** [comment]
@wenjiu100 我使用作者提供的代码也遇到了相同的问题，请问您修改了tuev数据处理的啥东西？

**Comment by LwkBuaa** [comment]
@wenjiu100 tuab数据集微调过程中，采用文中提供的参数，3个epoch后过拟合，训练集精度达到99%以上，测试集80左右，您遇到过这个问题吗？

**Comment by TheNorthS** [comment]
UserWarning: y_pred contains classes not in y_true
  warnings.warn("y_pred contains classes not in y_true")I also had the same problem. Have you resolved it?

**Comment by TheNorthS** [comment]
> 这个问题出现在 TUEV 数据集上，替换我自己的数据集也会出现
I also had the same problem. Have you resolved it?

**Comment by wenjiu100** [comment]
> @wenjiu100tuab数据集微调过程中，采用文中提供的参数，3个epoch后过拟合，训练集精度达到99%以上，测试集80左右，您遇到过这个问题吗？

Sorry, I've been a bit busy lately before I saw it, a lot of the settings in the code he gave you are none, false, you'll have to change them to true, it's really easy to get to the best results early using this code for training.

**Comment by wenjiu100** [comment]
> @wenjiu100tuab数据集微调过程中，采用文中提供的参数，3个epoch后过拟合，训练集精度达到99%以上，测试集80左右，您遇到过这个问题吗？

Sorry, I've been a bit busy lately before I saw it, a lot of the settings in the code he gave you are none, false, you'll have to change them to true, it's really easy to get to the best results early using this code for training.

**Comment by wenjiu100** [comment]
> > 这个问题出现在 TUEV 数据集上，替换我自己的数据集也会出现
> > I also had the same problem. Have you resolved it?

这是因为类内不平衡导致的，不会影响最终的准确率的。

**Comment by Zhreyu** [comment]
@wenjiu100, Were you still receiving this warning when the metrics began to align with those reported by the authors? Specifically, did you continue to see this UserWarning while the metrics appeared accurate?
